presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java

```diff
@@ -515,6 +517,13 @@ public void validate(OrcDataSource input)
                 orcEncoding);
     }
 
+    private int estimateChunkRowBoundByLogicalSize(Page page)
```

Comment:
I would make the function returns estimated row logical size (maybe called `estimateAverageLogicalSizePerRow`), and leave the row count calculation in `write`. Otherwise `chunkRowBoundByLogicalSize` doesn't look like an easy concept to understand.